### DISCUSSION TITLE
IKEv2 NAT-T broken after re-keying

### DISCUSSION THREAD

**Discussion by mmabln** [strongswan/strongswan#670] [Q&A] [answered]
Hi,

I am seeing a weird problem with a partner for site-to-site ikev2 vpn. After first negotiation (phase1+phase2) the tunnel comes up fine with NAT-T support, but after re-keying the remote side is sending raw ESP instead of UDP encapsulated ESP, so all ESP traffic from remote side gets dropped.

However I am seeing still NAT keepalive packets sent and received.

Is this a malfunction on remote side VPN gateway or is there anything which can be done on our side to fix this? The VPN was working fine for months and now shows this problem without anything changed on our side (we did changes to other vpns terminated on the same gateway, but no change for this specific vpn).

From Wireshark capture I can see this:

    No.  Time                         Source         Destination    Protocol  Length  Info
    161  2021-10-15  10:41:32,850205  172.27.253.15  11.2.123.1     ESP       178     ESP (SPI=0x92b5b35e)
    162  2021-10-15  10:41:32,861500  11.2.123.1     172.27.253.15  ESP       178     ESP (SPI=0xce96ce21)
    163  2021-10-15  10:41:52,137865  172.27.253.15  11.2.123.1     UDPENCAP  43      NAT-keepalive
    164  2021-10-15  10:41:53,184903  11.2.123.1     172.27.253.15  UDPENCAP  43      NAT-keepalive
    165  2021-10-15  10:42:12,138154  172.27.253.15  11.2.123.1     UDPENCAP  43      NAT-keepalive
    166  2021-10-15  10:42:13,219912  11.2.123.1     172.27.253.15  UDPENCAP  43      NAT-keepalive
    167  2021-10-15  10:42:20,110039  172.27.253.15  11.2.123.1     ISAKMP    654     CREATE_CHILD_SA MID=02 Initiator Request
    168  2021-10-15  10:42:20,173731  11.2.123.1     172.27.253.15  ISAKMP    526     CREATE_CHILD_SA MID=02 Responder Response
    169  2021-10-15  10:42:20,179609  172.27.253.15  11.2.123.1     ISAKMP    126     INFORMATIONAL MID=03 Initiator Request
    170  2021-10-15  10:42:20,201851  11.2.123.1     172.27.253.15  ISAKMP    126     INFORMATIONAL MID=03 Responder Response
    171  2021-10-15  10:42:32,910502  172.27.253.15  11.2.123.1     ESP       178     ESP (SPI=0xf103510c)
    172  2021-10-15  10:42:32,918210  11.2.123.1     172.27.253.15  ESP       170     ESP (SPI=0xc6ec91f9)
    173  2021-10-15  10:42:52,138886  172.27.253.15  11.2.123.1     UDPENCAP  43      NAT-keepalive
    174  2021-10-15  10:42:53,298407  11.2.123.1     172.27.253.15	UDPENCAP  43      NAT-keepalive

Packets 161+162 is an encrypted icmp echo-request (ping) packet, sent out as UDP encap. ESP, reply is also coming as UDP encap. ESP (port udp 4500).

Packet 167 to 170 is the re-keying.

Packet 171 is next icmp echo-request packet, sent out as UDP encap. ESP, but this time the reply coming back in packet 172 is raw ESP, not encapsulated in UDP (you can see this also from different packet length). From this point on it continues like this - all replies from remote side are coming as unencapsulated esp - and does not recover.

Only when tunnel is completely re-negotiated (down/up) it is working again until re-keying.

Log of charon for the re-keying process:

Oct 15 10:42:20 vpn-1 charon: 13[KNL] creating rekey job for CHILD_SA ESP/0x92b5b35e/11.2.123.1
Oct 15 10:42:20 vpn-1 charon: 13[IKE] establishing CHILD_SA xxxx_2{1}
Oct 15 10:42:20 vpn-1 charon: 13[ENC] generating CREATE_CHILD_SA request 2 [ N(REKEY_SA) SA No KE TSi TSr ]
Oct 15 10:42:20 vpn-1 charon: 13[NET] sending packet: from 172.27.253.15[4500] to 11.2.123.1[4500] (608 bytes)
Oct 15 10:42:20 vpn-1 charon: 06[NET] received packet: from 11.2.123.1[4500] to 172.27.253.15[4500] (480 bytes)
Oct 15 10:42:20 vpn-1 charon: 06[ENC] parsed CREATE_CHILD_SA response 2 [ SA No KE TSi TSr ]
Oct 15 10:42:20 vpn-1 charon: 06[IKE] CHILD_SA xxxx_2{96} established with SPIs c6ec91f9_i f103510c_o and TS 10.200.0.0/28 === 10.252.244.0/24
Oct 15 10:42:20 vpn-1 charon: 06[IKE] closing CHILD_SA xxxx_2{79} with SPIs ce96ce21_i (119147 bytes) 92b5b35e_o (11817 bytes) and TS 10.200.0.0/28 === 10.252.244.0/24
Oct 15 10:42:20 vpn-1 charon: 06[IKE] sending DELETE for ESP CHILD_SA with SPI ce96ce21
Oct 15 10:42:20 vpn-1 charon: 06[ENC] generating INFORMATIONAL request 3 [ D ]
Oct 15 10:42:20 vpn-1 charon: 06[NET] sending packet: from 172.27.253.15[4500] to 11.2.123.1[4500] (80 bytes)
Oct 15 10:42:20 vpn-1 charon: 09[NET] received packet: from 11.2.123.1[4500] to 172.27.253.15[4500] (80 bytes)
Oct 15 10:42:20 vpn-1 charon: 09[ENC] parsed INFORMATIONAL response 3 [ D ]
Oct 15 10:42:20 vpn-1 charon: 09[IKE] received DELETE for ESP CHILD_SA with SPI 92b5b35e
Oct 15 10:42:20 vpn-1 charon: 09[IKE] CHILD_SA closed

$ strongswan --version
Linux strongSwan U5.4.0/K4.14.238-125.422.amzn1.x86_64

This is an AWS EC2 instance with elastic ip, so for application it looks like 1:1 NAT.

Config of this connection:

config setup
        uniqueids=yes
        charondebug=
        charonstart=yes

conn xxxx_2
        right=11.2.123.1
        esp=aes256-sha256-modp2048
        auto=route
        rightsubnet=10.252.244.0/24
        ike=aes256-sha256-modp2048
        ikelifetime=28800s
        lifetime=3600s
        keyexchange=ikev2
        leftid=1.2.3.4 (our public ip)
        compress=no
        rightauth=psk
        leftauth=psk
        leftsubnet=10.200.0.0/28
        ikelifetime=28800s
        lifetime=3600s
        aggressive=no
        type=tunnel
        left=172.27.253.15

**Comment by tobiasbrunner** [comment, marked as answer]
> Is this a malfunction on remote side VPN gateway

Yes, it apparently disabled UDP encapsulation for the new CHILD_SA. Other than increasing the lifetime to avoid the rekeying, there is nothing you can do on your end.
